src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -582,11 +595,13 @@ def sort(self, cols: List[Expression]) -> "SelectStatement":
             new.from_ = self.from_.to_subqueryable()
             new.pre_actions = new.from_.pre_actions
             new.post_actions = new.from_.post_actions
-            new.order_by = cols
+            new.order_by = OrderByClause(cols, dependent_columns)
```

Comment:
This a is a bug that will be addressed by [SNOW-756443](https://snowflakecomputing.atlassian.net/browse/SNOW-756443), where chained sorts/orderby should preserve all the dependent columns in order.